### DISCUSSION TITLE
Intelliflo 3 RS485 Send Issues.

### DISCUSSION THREAD

**Discussion by truvec** [tagyoureit/nodejs-poolController#888] [Nixie]
I have nixie set up and have the pump going into automation mode. The pump can be controlled and nixie shows data from the pump. Pool control shows that all (100%) send stats are unsuccessful. I am using Mega-Bas but have also used usb rs485 with same effect.

**Comment by rstrouse** [comment]
Is the connection light turning Magenta on the pump?

**Comment by truvec** [comment]
Yes - it is. It sets speed or gpm according to the dashboard.

**Comment by truvec** [comment]
Just noticed something - If the Panel is controlling the pump via RPM it sets the speed. If set for GPM control it starts to set but then seems to reset and then loops.

**Reply by rstrouse** [reply]
Update your njsPC and lets see if it registers responses.  It looks like this pump always responds on the broadcast address.

**Comment by truvec** [comment]
That works much better. No aborts till . . .

Should we be able to have a rpm circuit and a gpm circuit?  

When engaging both it moves toward the higher but there seemed to be undefined responses(Aborts and retries).  Individually they play nice.

<img width="435" alt="Screenshot 2024-01-22 at 7 30 52 PM" src="https://github.com/tagyoureit/nodejs-poolController/assets/239790/2135e555-5086-433c-ad95-075ff17485c4">

Replay showing engaging rpm and gpm circuit.

Thank You!

**Reply by truvec** [reply]
Also, want to add if RPM is the higher speed it will run OK. if GPM is the higher circuit the aborts start stacking.

e.g.  Circuit1 = 30GPM + Circuit 2=2100 RPM this does not cause aborts. Circuit1=100GPM + Circuit2=1300RPM -> Aborts

**Reply by rstrouse** [reply]
While njsPC has an algorithm to handle it, the conversion to rpm is subjective and honestly the GPM calculator in pump is reliant on horsepower and wattage.  If I were you I would never look to do anything in GPM.  I did not see any aborts in your replay though.

In looking at your setup however, I think you are missing the point of the Pool circuit.  If the IntelliFlo 3 is your filter pump you should set a minimum speed  for the Pump to run when the pool circuit is engaged the Run circuit is not desired or required.  The minimum speed required is typically enough to trigger the flow switch for the chlorinator and a small amount of water movement.

The running speed of the pump is the one with the most flow from all of the currently running circuits and features.

**Reply by truvec** [reply]
Thanks. 

OK. Got it. The pool/spa circuit is the filter circuit. 

Here is the aborted message

[22/01/2024, 19:15:04] warn: Message aborted after 2 attempt(s): 165,0,96,33,10,4,2,196,9,107.81790774796491,2,110.81790774796491 
[22/01/2024, 19:15:04] error: Error sending setPumpRPMAsync for Pump 1: Message aborted after 2 attempt(s): 165,0,96,33,10,4,2,196,9,107.81790774796491,2,110.81790774796491

**Reply by rstrouse** [reply]
Do you have a spa that is connected to the pool?

**Reply by rstrouse** [reply]
Well that message is malformed.  I'll have a look.

165,0,96,33,10,4,2,196,9,107.81790774796491,2,110.81790774796491

It should be 165,0,96,33,10,4,2,196,9,108,2,111

**Reply by truvec** [reply]
No spa.

I was looking through the circuits today and saw there is a pool only circuit. Also different heat functions, (heat enable, heat boost, pool heater, spa heater, pool heat enable, etc). Also pump speed + Pump Speed - etc. Is there documentation to  describe the different use cases? I realize that the model follows pool industry automation nomenclature. i was thinking about perusing the varius automation panel manuals. 

The pool is only 3 weeks old. 

 Single body
- Intelliflo
- SWG - Hayward
- Acid pump
- Heat Cool Pump - Aquacal
- 2 water features 
- 6 microbrites on individual relays with one transformer(300w) (sync issues and light group not properly working)


Thank you.

**Reply by rstrouse** [reply]
Update your njsPC it should no longer be confused by the rpm to gpm and back conversion.  Also, if you have a pool/spa combo set the controller model to Nixie Shared Body

![image](https://github.com/tagyoureit/nodejs-poolController/assets/47839015/e9fb8bff-51f2-47ae-b171-c2a8c9844d43)

**Comment by truvec** [comment]
OK - No aborts.  

Still some funky behavior in control. I made a replay for the first instance.

1. Set low speed in RPM, higher speed in GPM. 1100rpm and 80 GPM. Does 1100 RPM. add the 2nd circuit. Runs at 96 GPM.

2. Set low speed in GPM, higher speed in RPM. 20GPM and 1900 RPM. Does 20 GPM. add the 2nd circuit. Runs at 1900RPM.

**Reply by rstrouse** [reply]
Hmmm... mathematically 1100 rpm will equate to around 36gpm and 80gpm is roughly equivalent to 2412rpm.  So we have some wires crossed in our algorithm.  I will take a look at what it is doing and let you know when I have it fixed.

**Reply by rstrouse** [reply]
Update njsPC, I found the issue.  It appears that it was matching against the wrong value.  I reworked the algorithm to choose between flow or rpm based on some additional factors.

**Reply by truvec** [reply]
Yes, it works well. I did test some near RPM - GPM speeds and the rpm would win out even though the reported GPM was lower.

eg
C1=1350 C2=40GPM 
C1+C2 -> DashPanel - GPM=37GPM RPM=1350.

 I don't think it's critical but wanted to let you know.

Thank you.

**Reply by rstrouse** [reply]
Yeah the power factors are not linear, We should be pretty close.

**Reply by truvec** [reply]
It's probably best to stick to either Flow or Speed (not both) when setting features.

**Reply by jaysalazar** [reply]
@truvec , I have been having trouble with a new Intelliflo 3 VSF pump. Can you confirm that your pump is working well in your setup? 
would you be kind enough to share your hardware/software setup?
